src/commonMain/kotlin/exchange.dydx.abacus/functional/vault/Vault.kt

```diff
@@ -94,13 +94,15 @@ object VaultCalculator {
         return parser.asTypedObject<IndexerMegavaultPositionResponse>(apiResponse)
     }
 
-    fun calculateVaultSummary(historical: IndexerMegavaultHistoricalPnlResponse?): VaultDetails? {
-        if (historical?.megavaultPnl.isNullOrEmpty()) {
+    fun calculateVaultSummary(historicals: List<IndexerMegavaultHistoricalPnlResponse>?): VaultDetails? {
```

Comment:
- processing it in the supervisor would be combining the raw response, doesn't seem like it fits
- I considered processing in the state machine , but the state machine is fairly straightforward adapter boilerplate so did not want to introduce uncommon logic there. e.g

```
internal fun TradingStateMachine.onMegaVaultPnl(
    payloads: List<String>
): StateChanges {
    val responses = payloads.mapNotNull { parser.asTypedObject<IndexerMegavaultHistoricalPnlResponse>(it) }
    val newState = vaultProcessor.processMegaVaultsHistoricalPnl(internalState.vault, responses)
    return updateVaultState(internalState, newState)
}

internal fun TradingStateMachine.onVaultMarketPnls(
    payload: String
): StateChanges {
    val pnlResponse = parser.asTypedObject<IndexerVaultsHistoricalPnlResponse>(payload)
    val newState = vaultProcessor.processVaultMarketHistoricalPnls(internalState.vault, pnlResponse)
    return updateVaultState(internalState, newState)
}

internal fun TradingStateMachine.onVaultMarketPositions(
    payload: String
): StateChanges {
    val positionResponse = parser.asTypedObject<IndexerMegavaultPositionResponse>(payload)
    val newState = vaultProcessor.processVaultMarketPositions(internalState.vault, positionResponse)
    return updateVaultState(internalState, newState)
}
```

Reply:
yeah it does seem like the state machine is the best place. I am not familiar with that code and haven't followed much, are you saying we'd have to duplicate the combining logic if we put it there?

Reply:
we would be manipulating the response objects in order to combine them into a single object which feels like it should be done in the calculator instead of the state machine

Reply:
so to clarify, I feel differently about this:
> it does seem like the state machine is the best place